system/ear/jsps/rxlogin.jsp

```diff
@@ -3,38 +3,14 @@
 <%@ page import="com.percussion.i18n.PSI18nUtils" %>
 <%@ page import="com.percussion.i18n.PSLocaleManager" %>
 <%@ page import="com.percussion.i18n.PSLocale" session="true" %>
-<%@ page import="com.percussion.security.SecureStringUtils" %>
+<%@ page import="com.percussion.i18n.PSLocaleException" %>
 <%@ taglib uri="http://www.owasp.org/index.php/Category:OWASP_CSRFGuard_Project/Owasp.CsrfGuard.tld" prefix="csrf" %>
 <%@ taglib uri="http://rhythmyx.percussion.com/components"
 		   prefix="rxcomp"%>
-<%--
-  ~     Percussion CMS
-  ~     Copyright (C) 1999-2021 Percussion Software, Inc.
-  ~
-  ~     This program is free software: you can redistribute it and/or modify
-  ~     it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
-  ~
-  ~     This program is distributed in the hope that it will be useful,
-  ~     but WITHOUT ANY WARRANTY; without even the implied warranty of
-  ~     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-  ~     GNU Affero General Public License for more details.
-  ~
-  ~     Mailing Address:
-  ~
-  ~      Percussion Software, Inc.
-  ~      PO Box 767
-  ~      Burlington, MA 01803, USA
-  ~      +01-781-438-9900
-  ~      support@percussion.com
-  ~      https://www.percussion.com
-  ~
-  ~     You should have received a copy of the GNU Affero General Public License along with this program.  If not, see <https://www.gnu.org/licenses/>
-  --%>
-
 <%
 	//Checking for vulnerability
 	String str = request.getQueryString();
-	if(str != null && str != ""){
+	if(str == null || str.trim().equals("")){
```

Comment:
not understading this change, as it is totally opposite now.

Reply:
I left that in based on the comment, but it doesn't make sense that it would forbid the request if there was a query string, as we always have a query string.  with the redirectURL and the OWASP token.  I think if there ISN't a query string that would be more of an issue.  I did test the login and it does seem to work ok.

Reply:
I almost just want to take that out completely as it is unclear what vulnerability that if block is fixing.

Reply:
am taking that block out.